docs/topics/compatibility-guide-15.md

```diff
@@ -0,0 +1,274 @@
+
+### Forbid exposing anonymous types from private inline functions
+
+> **Issue**: [KT-33917](https://youtrack.jetbrains.com/issue/KT-33917)
+>
+> **Component**: Core language
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: Kotlin 1.5 will outlaw exposing anonymous types from private inline functions
+>
+> **Deprecation cycle**:
+>
+> - < 1.5: introduce warning for the problematic constructors
+> - \>= 1.5: raise this warning to an error,
+>  `-XXLanguage:-ApproximateAnonymousReturnTypesInPrivateInlineFunctions` can be used to temporarily revert to pre-1.5 behavior
+
+### Support special semantics for underscore-named catch block parameters
+
+> **Issue**: [KT-31567](https://youtrack.jetbrains.com/issue/KT-31567)
+>
+> **Component**: Core language
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: Kotlin 1.5 will outlaw references to the underscore symbol (`_`) that is used to omit parameter name of an exception in the catch block
+>
+> **Deprecation cycle**:
+>
+> - 1.4.20: introduce warning for the problematic references
+> - \>= 1.5: raise this warning to an error,
+>  `-XXLanguage:-ForbidReferencingToUnderscoreNamedParameterOfCatchBlock` can be used to temporarily revert to pre-1.5 behavior
+
+### It's possible to pass non-spread arrays after arguments with SAM-conversion
+
+> **Issue**: [KT-35224](https://youtrack.jetbrains.com/issue/KT-35224)
+>
+> **Component**: Core language
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: Kotlin 1.5 will outlaw passing non-spread arrays after arguments with SAM-conversion
+>
+> **Deprecation cycle**:
+>
+> - 1.3.70: introduce warning for the problematic calls
+> - \>= 1.5: raise this warning to an error,
+>  `-XXLanguage:-ProhibitVarargAsArrayAfterSamArgument` can be used to temporarily revert to pre-1.5 behavior
+
+### Change implementation strategy of SAM conversion from anonymous class-based to invokedynamic
+
+> **Issue**: [KT-44912](https://youtrack.jetbrains.com/issue/KT-44912)
+>
+> **Component**: Kotlin/JVM
+>
+> **Incompatible change type**: behavioral
+>
+> **Short summary**: Since Kotlin 1.5, implementation strategy of SAM (single abstract method) conversion will be changed from generating an anonymous class to using the `invokedynamic` JVM instruction
+>
+> **Deprecation cycle**:
+>
+> - 1.5: change implementation strategy of SAM conversion,
+>  `-Xsam-conversions=class` can be used to revert implementation scheme to the one that used before
+
+### Generate nullability assertion for delegated property with generic call in the delegate expression
+
+> **Issue**: [KT-44304](https://youtrack.jetbrains.com/issue/KT-44304)
+>
+> **Component**: Kotlin/JVM
+>
+> **Incompatible change type**: behavioral
+>
+> **Short summary**: Since Kotlin 1.5, the Kotlin compiler will emit nullability assertion for delegated property with generic call in the delegate expression
```

Comment:
```suggestion
> **Short summary**: Since Kotlin 1.5, the Kotlin compiler will emit nullability assertions for delegated properties with a generic call in the delegate expression
```